### ISSUE TITLE
Desktop Apps

### ISSUE THREAD

**Issue by TCB13** [BaldissaraMatheus/Tasks.md#68]
This is great! Thank you.

Why not turn it into a Desktop app via Electron or a similar solution? Since it stores everything as plain text files a desktop app would be great because people could just download it and use it standalone (easier than docker + browser) and sync files between machines using some cloud service or Synching.

**Comment by BaldissaraMatheus** [comment]
To be honest this is something that I never considered, but it's not a bad idea at all! I'm not very used to developing desktop apps, so I don't really know how much work I'd need to put into it port the app, but I can look into it!

**Comment by jcvanhalle-ec** [comment]
> Why not turn it into a Desktop app via Electron or a similar solution? Since it stores everything as plain text files a desktop app would be great because people could just download it and use it standalone (easier than docker + browser) and sync files between machines using some cloud service or Synching.

Yeah. But. I use it *because* it's in the browser and I can self host it (either for sharing or for access from any devices). A desktop app is a different paradigm.

**Comment by BaldissaraMatheus** [comment]
As an alternative, you can install it as a chrome PWA. I know it's not really the same thing, but it's better than nothing

**Comment by TCB13** [comment]
> > Why not turn it into a Desktop app via Electron or a similar solution? Since it stores everything as plain text files a desktop app would be great because people could just download it and use it standalone (easier than docker + browser) and sync files between machines using some cloud service or Synching.
> 
> Yeah. But. I use it _because_ it's in the browser and I can self host it (either for sharing or for access from any devices). A desktop app is a different paradigm and implies different requirements (installed libs, OSes, etc.) and delivery mechanisms (appimage, flak, deb., etc.) and test coverage.

You're completely missing the point and over exaggerating the task.

If you want to have as it is, fine, self host it somewhere and use that way. The thing with having this packaged as a desktop application is that this is well built and works around markdown files that can be easily synced between machines thus removing the need for a central "server".

**Comment by jcvanhalle-ec** [comment]
> You're completely missing the point and over exaggerating the task.

I am not missing your point. I am not saying it's hard, I am saying it's something different to handle. I am also saying that, and I quote, `turn[ing] it into a Desktop app via Electron or a similar solution`, is a different use case and scenario from how the app can be used and deployed at the moment and that I'd rather for it to keep on being possible to host it.

> If you want to have as it is, fine, self host it somewhere and use that way. [..]

Thank you for allowing me to do that :].

>  The thing with having this packaged as a desktop application is that this is well built and works around markdown files that can be easily synced between machines thus removing the need for a central "server".

Well, when it's hosted you don't need syncing.

Anyway, it looks like I hit a nerve or something or I came across as insulting, sorry about that, but I stand by my point that a desktop app (however cool or needed or easy, etc.) is not the same thing as a hosted app and that it's packaged differently and I fail to see what's controversial about that.

**Comment by BaldissaraMatheus** [comment]
Sorry, I decided to not work on it. I don't plan to increase the scope of the project for now, so I'm closing this issue.

**Comment by multimeric** [comment]
Would you consider a PR adding an Electron (etc) app build?

**Comment by BaldissaraMatheus** [comment]
@multimeric yes
